evalml/pipelines/components/transformers/encoders/label_encoder.py

```diff
@@ -46,6 +47,7 @@ def fit(self, X, y):
         if y is None:
             raise ValueError("y cannot be None!")
         y_ww = infer_feature_types(y)
+        self.original_typing = str(y_ww.ww.logical_type)
```

Comment:
@Cmancuso I think we need to keep this functionality as long as woodwork transforms 1/0, yes/no etc. to True/False unless that change was made for typelib

Reply:
I thought we weren't seeing this issue in schemaUpdate?

Reply:
on the EvalML side we'd still want to output the original form of the target instead of outputting True/False if the target is boolean or boolean inferable.

Reply:
Yeah, this will only be an issue for OS users when they use it in this specific scenario. If users pass in a `yes/no` dataset, they should still receive `yes/no` predictions, so I think this is needed.